src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -299,17 +305,19 @@ internal TensorFlowTransform(IHostEnvironment env, Arguments args, IDataView inp
 
             if (args.ReTrain)
             {
-                // Shuffling fails because ShuffleTransform is not RowToRowMapper
-                // https://github.com/dotnet/machinelearning/issues/1106
-                //if(args.Shuffle)
-                //{
-                //    input = new ShuffleTransform(env, new ShuffleTransform.Arguments(), input);
-                //}
-
                 env.CheckValue(input, nameof(input));
 
                 CheckTrainingParameters(args);
 
+                if(args.Shuffle)
```

Comment:
>Shuffle [](start = 24, length = 7)

how is this bypassing the row mapper problem?

Reply:
This is happening at training time. The row mapper problem is happening at `CreatePredictionEngine` time. Please see #1106 for more info.

---
In reply to: [224640317](https://github.com/dotnet/machinelearning/pull/1208#discussion_r224640317) [](ancestors = 224640317)